### DISCUSSION TITLE
Write preferences to a file

### DISCUSSION THREAD

**Discussion by rohan-patra** [coder/code-server#3790] [Q&A] [answered]
I am running vs code server in a docker container, and I was wondering if there was any way to write all my vscode preferences to a specific file automatically so that I can write that file to my dockerfile and my preferences and extensions can persist between redeploys?

**Comment by jsjoeio** [comment, marked as answer]
Yes! We have some docs [here](https://coder.com/docs/code-server/v3.11.0/FAQ#how-can-i-reuse-my-vs-code-configuration) about how to do that. If you get stuck or need more help, let us know!

**Reply by rohan-patra** [reply]
Thanks for the response!

I was actually wondering if there is a way to specify where vscode should write preferences to. In the link you sent, all I am able to do is specify where vscode should read preferences from.

**Reply by jsjoeio** [reply]
Hmm...I may be wrong here but I assume when you use `--user-data-dir <dir>` that says "this is where you read/write preferences". Is that not the case?

**Reply by rohan-patra** [reply]
From what I can tell, it is just reading the data you manually save in <dir> and copying it over to ~/.local/share/code-server which it continues to use for reading and writing preferences.

**Reply by jsjoeio** [reply]
Ahh okay. And you want it to stay in `<dir>` instead of copying over? Is that right?

**Reply by rohan-patra** [reply]
yep

**Reply by jsjoeio** [reply]
Got it! Let me double-check with @code-asher - is this the intended behavior? What are your thoughts on this?

**Reply by code-asher** [reply]
`--user-data-dir` will cause VS Code to read and write all settings to the provided directory.

Keep in mind that it won't move the old directory so if you already have `~/.local/share/code-server` it will continue to exist (but code-server will neither read from nor write to it).

If it's not doing that then it's a bug! Please feel free to open an issue with steps to reproduce.

**Reply by rohan-patra** [reply]
I just found out my issue, I had an extra slash at the beginning of the directory I was passing and it was getting saved in my root directory which I was not able to add to my docker image. Once I specified the correct directory, everything worked perfectly!

Thank you both for your help @jsjoeio @code-asher
